src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String EXCEPTION = "Wrong index - ";
+    private Node first;
+    private Node last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node newNode = new Node(value);
+        if (size == 0) {
+            first = newNode;
+            last = newNode;
+        } else {
+            newNode.prev = last;
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException(EXCEPTION + index);
+        }
```

Comment:
This could be extracted to a separate method.

Reply:
I can't put this check into a general method because it checks a different range of values. index > size

Reply:
Make the separate method as for get and remove.